ocp_resources/resource.py

```diff
@@ -1122,13 +1122,15 @@ def api_request(self, method: str, action: str, url: str, **params: Any) -> Dict
 
         """
         client: DynamicClient = self.privileged_client or self.client
-        response = client.client.request(
+        response = self.retry_cluster_exceptions(
```

Comment:
this adds 30 seconds for every API call, we should not do this.

Reply:
This is only used in vm and vmi calls 
```
cp_resources/resource.py:    def api_request(self, method: str, action: str, url: str, **params: Any) -> Dict[str, Any]:
ocp_resources/virtual_machine.py:    def api_request(self, method, action, **params):
ocp_resources/virtual_machine.py:        return super().api_request(method=method, action=action, url=self._subresource_api_url, **params)
ocp_resources/virtual_machine.py:        self.api_request(method="PUT", action="start")
ocp_resources/virtual_machine.py:        self.api_request(method="PUT", action="restart")
ocp_resources/virtual_machine.py:        self.api_request(method="PUT", action="stop")
ocp_resources/virtual_machine_instance.py:    def api_request(self, method, action, **params):
ocp_resources/virtual_machine_instance.py:        return super().api_request(method=method, action=action, url=self._subresource_api_url, **params)
ocp_resources/virtual_machine_instance.py:        self.api_request(method="PUT", action="pause")
ocp_resources/virtual_machine_instance.py:        self.api_request(method="PUT", action="unpause")
ocp_resources/virtual_machine_instance.py:        return self.api_request(method="GET", action="guestosinfo")
ocp_resources/virtual_machine_instance.py:        return self.api_request(method="GET", action="filesystemlist")
ocp_resources/virtual_machine_instance.py:        return self.api_request(method="GET", action="userlist")
dbasunag@dbasunag-mac openshift-python-wrapper % 
```

Reply:
Added a dict with timeout and sleep_time value as you suggested.